python/ray/air/_internal/remote_storage.py

```diff
@@ -95,15 +95,22 @@ def get_fs_and_path(
         return fs, path
 
     try:
-        fs, path = pyarrow.fs.FileSystem.from_uri(uri)
-        _cached_fs[cache_key] = fs
-        return fs, path
+        # In case of hdfs filesystem, if uri does not have the netloc part below will
+        # failed with hdfs access error.  For example 'hdfs:///user_folder/...' will
+        # fail, while only 'hdfs://namenode_server/user_foler/...' will work
+        if not (parsed.scheme == "hdfs" and parsed.netloc == ""):
+            fs, path = pyarrow.fs.FileSystem.from_uri(uri)
+            _cached_fs[cache_key] = fs
+            return fs, path
     except (pyarrow.lib.ArrowInvalid, pyarrow.lib.ArrowNotImplementedError):
         # Raised when URI not recognized
         if not fsspec:
             # Only return if fsspec is not installed
             return None, None
 
+    if parsed.scheme == "hdfs" and parsed.netloc == "" and not fsspec:
```

Comment:
It could be cleaner if moving this check above the try block, and then inside the try block there is no need to check this condition again.